compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -490,22 +490,23 @@ impl<'a, 'tcx> EncodeContext<'a, 'tcx> {
                     FileName::Real(ref realname) => {
                         let mut adapted = (**source_file).clone();
                         adapted.name = FileName::Real(match realname {
-                            RealFileName::LocalPath(local_path) => {
-                                // Prepend path of working directory onto local path.
-                                // because relative paths are potentially relative to a
-                                // wrong directory.
+                            RealFileName::LocalPath(relative) => {
```

Comment:
We actually don't know if this is a relative path or an absolute one. It's just that `Path::join` is a no-op if the argument is an absolute path, so we don't care.